packages/eslint-config-eslint/tests/types/types.test-d.mts

```diff
@@ -0,0 +1,20 @@
+import type { Linter } from "eslint";
+
+import * as eslintConfigESLint from "eslint-config-eslint";
+import * as eslintConfigESLintBase from "eslint-config-eslint/base";
+import * as eslintConfigESLintCJS from "eslint-config-eslint/cjs";
+import * as eslintConfigESLintFormatting from "eslint-config-eslint/formatting";
```

Comment:
> I'm not sure how these tests are passing because, for example, `eslintConfigESLint` evaluates to `{ default: [...] }`.

I think that's because the [`allowSyntheticDefaultImports`](https://www.typescriptlang.org/tsconfig/#allowSyntheticDefaultImports) is set to `false` in the TypeScript config. We could change that option to `true` to disallow the `import *` syntax in the types tests if it's confusing.

Reply:
> I think that's because the [`allowSyntheticDefaultImports`](https://www.typescriptlang.org/tsconfig/#allowSyntheticDefaultImports) is set to `false` in the TypeScript config. We could change that option to `true` to disallow the `import *` syntax in the types tests if it's confusing.

Sounds good to me 👍